### PR TITLE
Add BCP47 parser

### PR DESCRIPTION
**Pull request by miniBill**

### CODE REVIEW

**Comment by miniBill** [comment]
The type that is currently called `Country` and described as `ISO 3166-1 country code` is called `region` in BCP47, because it could also be an UM M49 [1] numeric code.

My proposal is to rename it to `Region`.

[1] https://en.wikipedia.org/wiki/UN_M49

EDIT: this has now been implemented in #3

File: src/LanguageTag/PrivateUse.elm
Comment (dillonkearns) [inline src/LanguageTag/PrivateUse.elm]: If I'm reading the spec correctly here, it looks like a private use tag needs to be 1-8 characters to be valid?

> composed of the letter x and a hyphen followed by subtags of one to eight characters each, separated by hyphens

https://en.wikipedia.org/wiki/IETF_language_tag#Syntax_of_language_tags

If so, it would be nice to validate that and only return a `PrivateUse` value if it is validated.

File: src/LanguageTag/Parser.elm
Comment (dillonkearns) [inline src/LanguageTag/Parser.elm:19]: Could you add some `elm-verify-examples` in the docs for `parseLanguageTag` and `parseBcp47`? That would be helpful for making users more confident that they're using the right function for their use case.

**Comment by dillonkearns** [comment]
Looks good other than the two comments I made and the `elm-review` errors in the CI. I'll merge and release once that's resolved. Thanks for the PR @miniBill!

**Comment by miniBill** [comment]
The `NoModuleOnExposedNames` errors are false positives, I can suppress them?

You said you wrote two comments but I can't see them?

With regards to releasing, I think it's valuable to first decide whether to merge #3 in the next major bump or not

**Comment by dillonkearns** [comment]
@miniBill is this the problem that's causing the false positives? https://github.com/sparksp/elm-review-imports/issues/127

I don't like the idea of using suppressed errors for bugs - I try to avoid suppressing errors, I see the main use case for those as being adopting a rule in a large codebase where you can't fix everything at once. I'd rather turn off the rule temporarily with a comment that links to the GitHub issue and explains that it can be re-enabled once that fixed. Would you be up for adding that in this PR?

> With regards to releasing, I think it's valuable to first decide whether to merge https://github.com/dillonkearns/elm-bcp47-language-tag/pull/3 in the next major bump or not

Good idea, I'll hold off on releasing until we've discussed that.

**Comment by dillonkearns** [comment]
> You said you wrote two comments but I can't see them?

I can see them from this page <https://github.com/dillonkearns/elm-bcp47-language-tag/pull/2>.

I see the comments like this in that page:
Screenshot 2023-09-27 at 8 13 26 PM

**Comment by miniBill** [comment]
(notice how it says "Pending" on top, it means they are not visible to me yet)
Both comments make sense, will do!

**Review by dillonkearns — CHANGES_REQUESTED** [review]

**Comment by dillonkearns** [comment]
> (notice how it says "Pending" on top, it means they are not visible to me yet)

Ah yes, sorry about that, I sent them off for review so should be good now!

File: src/LanguageTag/PrivateUse.elm
Comment (miniBill) [reply src/LanguageTag/PrivateUse.elm]: Makes sense

**Comment by miniBill** [comment]
Made the changes requested

**Comment by dillonkearns** [comment]
Looks good, thank you for the PR @miniBill!

I'll let you follow up with #3 before I cut a new release.

**Pull request merged**
